src/hooks/screen/useCreatePostHook.js

```diff
@@ -12,27 +12,22 @@ import AnonUserInfoRepo from '../../service/repo/anonUserInfoRepo';
 const useCreatePostHook = (isAnonymous) => {
   const {params = {}} = useRoute();
   const {topic} = params;
+  const [selectedTopic, setSelectedTopic] = React.useState(topic);
   const [anonUserInfo, setAnonUserInfo] = React.useState(null);
+  const [headerTitle, setHeaderTitle] = React.useState(
+    selectedTopic ? `Create Post in #${selectedTopic}` : 'Create Post'
+  );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `headerTitle` state is initialized with a value that depends on the `selectedTopic` state. However, at this point in the code, `selectedTopic` is always going to be `topic` because `setSelectedTopic` has not been called yet. This could lead to confusion and potential bugs in the future. It would be clearer to initialize `headerTitle` based directly on `topic`.

```diff
- const [headerTitle, setHeaderTitle] = React.useState(
-   selectedTopic ? `Create Post in #${selectedTopic}` : 'Create Post'
- );
+ const [headerTitle, setHeaderTitle] = React.useState(
+   topic ? `Create Post in #${topic}` : 'Create Post'
+ );
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->